spring-integration-websocket/src/main/java/org/springframework/integration/websocket/inbound/WebSocketInboundChannelAdapter.java

```diff
@@ -224,39 +224,54 @@ public List<String> getSubProtocols() {
 	}
 
 	@Override
-	public void afterSessionStarted(WebSocketSession session) throws Exception { // NOSONAR Thrown from the delegate
+	public void afterSessionStarted(WebSocketSession session) {
 		if (isActive()) {
-			SubProtocolHandler protocolHandler = this.subProtocolHandlerRegistry.findProtocolHandler(session);
-			protocolHandler.afterSessionStarted(session, this.subProtocolHandlerChannel);
-			if (!this.server && protocolHandler instanceof StompSubProtocolHandler) {
-				StompHeaderAccessor accessor = StompHeaderAccessor.create(StompCommand.CONNECT);
-				accessor.setSessionId(session.getId());
-				accessor.setLeaveMutable(true);
-				accessor.setAcceptVersion("1.1,1.2");
-
-				Message<?> connectMessage = MessageBuilder.createMessage(EMPTY_PAYLOAD, accessor.getMessageHeaders());
-				protocolHandler.handleMessageToClient(session, connectMessage);
+			try {
+				SubProtocolHandler protocolHandler = this.subProtocolHandlerRegistry.findProtocolHandler(session);
+				protocolHandler.afterSessionStarted(session, this.subProtocolHandlerChannel);
+				if (!this.server && protocolHandler instanceof StompSubProtocolHandler) {
+					StompHeaderAccessor accessor = StompHeaderAccessor.create(StompCommand.CONNECT);
+					accessor.setSessionId(session.getId());
+					accessor.setLeaveMutable(true);
+					accessor.setAcceptVersion("1.1,1.2");
+
+					Message<?> connectMessage =
+							MessageBuilder.createMessage(EMPTY_PAYLOAD, accessor.getMessageHeaders());
+					protocolHandler.handleMessageToClient(session, connectMessage);
+				}
+			}
+			catch (Exception ex) {
+				logger.error(ex, () -> "WebSocketHandler.afterConnectionEstablished threw exception in " + this);
 			}
 		}
 	}
 
 	@Override
-	public void afterSessionEnded(WebSocketSession session, CloseStatus closeStatus)
-			throws Exception { // NOSONAR Thrown from the delegate
+	public void afterSessionEnded(WebSocketSession session, CloseStatus closeStatus) {
 
 		if (isActive()) {
-			this.subProtocolHandlerRegistry.findProtocolHandler(session)
-					.afterSessionEnded(session, closeStatus, this.subProtocolHandlerChannel);
+			try {
+				this.subProtocolHandlerRegistry.findProtocolHandler(session)
+						.afterSessionEnded(session, closeStatus, this.subProtocolHandlerChannel);
+			}
+			catch (Exception ex) {
```

Comment:
I think you'll need `// NOSONAR' for these to avoid Sonar complaining.

Reply:
Why? What is that rule we would violate otherwise?

Reply:
I might be confusing it with `catch (Throwable t)`; I guess leave as is.

Reply:
I see. Probably you mean then this: https://rules.sonarsource.com/java/tag/error-handling/RSPEC-1181, which indeed doesn't talk about `catch Exception`